docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+- When tedge-agent is started, it will publish health status as `up` on to `tedge/health/tedge-agent` topic
+- Now the mapper picks up this message and creates the supported operations file (c8y_SoftwareUpdate, etc) in `/etc/tedge/operations/c8y` with their content. 
+- Mapper sends the updated supported operations list to the c8y cloud.
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When c8y-log-plugin is started, it will publish health status as `up` on to `tedge/health/c8y-log-plugin` topic
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management) to the c8y cloud.
+
+Pros: 
+-   This will remove the file system dependency when the daemons want to create the supported operation files on the mapper file system.
+    For example, tedge-agent wants to create the supported operations files in /etc/tedge/operations/c8y directory.
+
+Cons: 
+-   This leads to hard dependency on the list of daemons in the mapper. For example, the tedge-mapper-c8y has to subscribe to tedge/health/tedge-agent
+topic to receive the health status from the tedge-agent. Based on this the mapper has to create a supported operation file in /etc/tedge/operations/c8y.
+-   This will be a hard-coded dependency of the name of the plugin and can’t be updated on the fly if someone wants to change the name of the daemon
+    or wants to add a new daemon.
+
+## Proposal 2:  Introduce an `init message` mechanism
+
+Make all the tedge components publish a `retained init` message describing their role (software management, config management, etc)
+ and other details specific to that role.
+
+The proposed topic will be `tedge/<device-id>/init/<cloud>`.
+Here `<cloud>` could be c8y, azure, aws, etc
+
+The message template is {Operation_name: <content>, Operation_name:<content>, Config_types: type1..typen}.
+Here `Operation_name` is the name of the operation to be published to the cloud and 
+the `content` is the content of the operation file, if any.
+Here `Config_types` is the configuration types that are supported by that particular daemon.
+
+Example 1:
+- When the tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
+- When tedge-agent is started, it will publish `{}` an empty `init` message on to `tedge/init/c8y/tedge-agent`
```

Comment:
As mentioned earlier, `tedge-agent` which is a cloud agnsotic component, must not publish to the cloud specific`tedge/init/c8y` topic. And it must not be an empty message either. The `tedge-agent` must explicitly declare that it supports `software_list`, `software_update` and `device_restart` operations, which the mapper can map to `c8y_SoftwareUpdate`, `c8y_SoftwareList` and `c8y_Restart` operations.

Reply:
Yes, agreed. Updated spec.